cmd/dagger/main.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/dagger/dagger/engine"
+	"github.com/dagger/dagger/tracing"
+	"github.com/spf13/cobra"
+)
+
+var (
+	configPath string
+	workdir    string
+
+	queryFile      string
+	queryVarsInput []string
+
+	listenAddress string
+
+	disableHostRW bool
+	queryDebug    bool
+)
+
+func init() {
+	rootCmd.PersistentFlags().StringVar(&workdir, "workdir", ".", "The host workdir loaded into dagger")
+	rootCmd.AddCommand(
+		listenCmd,
+		versionCmd,
+		queryCmd,
+		execCmd,
+	)
+
+	// changing usage template so examples are displayed below flags
+	queryCmd.SetUsageTemplate(`Usage:{{if .Runnable}}
+  {{.UseLine}}{{end}}{{if .HasAvailableSubCommands}}
+  {{.CommandPath}} [command]{{end}}{{if gt (len .Aliases) 0}}
+
+Aliases:
+  {{.NameAndAliases}}{{end}}{{if .HasAvailableSubCommands}}{{$cmds := .Commands}}{{if eq (len .Groups) 0}}
+
+Available Commands:{{range $cmds}}{{if (or .IsAvailableCommand (eq .Name "help"))}}
+  {{rpad .Name .NamePadding }} {{.Short}}{{end}}{{end}}{{else}}{{range $group := .Groups}}
+
+{{.Title}}{{range $cmds}}{{if (and (eq .GroupID $group.ID) (or .IsAvailableCommand (eq .Name "help")))}}
+  {{rpad .Name .NamePadding }} {{.Short}}{{end}}{{end}}{{end}}{{if not .AllChildCommandsHaveGroup}}
+
+Additional Commands:{{range $cmds}}{{if (and (eq .GroupID "") (or .IsAvailableCommand (eq .Name "help")))}}
+  {{rpad .Name .NamePadding }} {{.Short}}{{end}}{{end}}{{end}}{{end}}{{end}}{{if .HasAvailableLocalFlags}}
+
+Flags:
+{{.LocalFlags.FlagUsages | trimTrailingWhitespaces}}{{end}}{{if .HasAvailableInheritedFlags}}{{if .HasExample}}
+
+Examples:
+{{.Example}}{{end}}
+Global Flags:
+{{.InheritedFlags.FlagUsages | trimTrailingWhitespaces}}{{end}}{{if .HasHelpSubCommands}}
+
+Additional help topics:{{range .Commands}}{{if .IsAdditionalHelpTopicCommand}}
+  {{rpad .CommandPath .CommandPathPadding}} {{.Short}}{{end}}{{end}}{{end}}{{if .HasAvailableSubCommands}}
+
+Use "{{.CommandPath}} [command] --help" for more information about a command.{{end}}
+`)
+
+	queryCmd.Flags().StringVarP(&queryFile, "doc", "", "", "document query file")
```

Comment:
nit (for another time, I know this is from cloak): we should define flags where they are used rather than all in main

Reply:
Oh, I actually found it nicer to have them all in `main`. No strong opinions though, we can refactor this.

Reply:
That's exactly what I mentioned yesterday.

I've seen the worst of what happens when this grows over the years, and while I am sure that this situation will be different, I would much rather see command flags living with the command source than everything in `main`. We don't need to do it in this PR - keep 'em small, keep 'em flowing - but we should address it within the next few weeks.

Reply:
fixed @gerhard. It wasn't a big change at all, it's ok to scope it to this PR.